app/controllers/case_management/tax_returns_controller.rb

```diff
@@ -23,17 +24,84 @@ def update
       redirect_to case_management_clients_path, notice: success_message
     end
 
-    def edit_status; end
+    def edit_status
+      # prefill status if we received a param
+      preselected_status = params.dig(:tax_return, :status)
+
+      @take_action_form = CaseManagement::TakeActionForm.new(
+        @client,
+        status: preselected_status || @tax_return.status,
+        locale: @client.intake.locale,
+        message_body: status_macro(preselected_status),
+        contact_method: preferred_contact_method_or_default
+      )
+    end
 
     def update_status
-      if @tax_return.update(status_params)
-        SystemNote.create_status_change_note(current_user, @tax_return)
-        redirect_to case_management_client_messages_path(client_id: @client.id)
+      @take_action_form = CaseManagement::TakeActionForm.new(@client, take_action_form_params)
+      if @take_action_form.valid?
+        action_list = []
+        if @take_action_form.status != @tax_return.status
+          @tax_return.update!(status: @take_action_form.status)
+          SystemNote.create_status_change_note(current_user, @tax_return)
+          action_list << I18n.t("case_management.tax_returns.edit_status.flash_message.status")
+        end
+
+        if @take_action_form.message_body.present?
+          case @take_action_form.contact_method
+          when "email"
+            send_email(@client, body: @take_action_form.message_body, subject_locale: @take_action_form.locale)
+            action_list << I18n.t("case_management.tax_returns.edit_status.flash_message.email")
+          when "text_message"
+            send_text_message(@client, body: @take_action_form.message_body)
+            action_list << I18n.t("case_management.tax_returns.edit_status.flash_message.text_message")
+          end
+        end
+
+        if @take_action_form.internal_note.present?
+          Note.create!(
+            body: @take_action_form.internal_note,
+            client: @client,
+            user: current_user
+          )
+          action_list << I18n.t("case_management.tax_returns.edit_status.flash_message.internal_note")
+        end
+
+        flash[:notice] = I18n.t("case_management.tax_returns.edit_status.flash_message.success", action_list: action_list.join(", ").capitalize)
+
+        redirect_to case_management_client_path(id: @client)
       end
     end
 
     private
 
+    def preferred_contact_method_or_default
+      default = "email"
+      prefers_sms_only = @client.intake.sms_notification_opt_in_yes? && @client.intake.email_notification_opt_in_no?
+      prefers_sms_only ? "text_message" : default
+    end
+
+    def status_macro(status)
+      case status
+      when "intake_more_info", "prep_more_info", "review_more_info"
+        document_list = @client.intake.relevant_document_types.map do |doc_type|
+          "  - " + doc_type.translated_label(@client.intake.locale)
```

Comment:
Change to form locale

Reply:
this turned out to be trickier than I expected. I'm going to defer until later